src/Middleware/ConvertErrorToJsonResponse.php

```diff
@@ -24,41 +24,60 @@ class ConvertErrorToJsonResponse
      * @return ResponseInterface
      */
     public function __invoke($error, ServerRequestInterface $request, ResponseInterface $response, callable $next): ResponseInterface
+    {
+        return new JsonResponse(self::buildErrorBody($error), self::getStatusCode($error));
+    }
+
+    public static function buildErrorBody($error): array
     {
         switch ($error) {
             case $error instanceof DomainException:
-                return $this->convertDomainException($error);
+                return self::buildDomainExceptionBody($error);
             case $error instanceof Throwable:
-                return $this->convertThrowable($error);
+                return self::buildThrowableBody($error);
             default:
-                return $this->convertGenericError($error);
+                return self::buildGenericErrorBody($error);
         }
     }
 
-    private function convertGenericError($error): JsonResponse
+    public static function getStatusCode($error): int
     {
-        $body = [
-            'code' => ExceptionTokens::AN_ERROR_HAS_OCCURRED,
-            'message' => 'A unexpected error has occurred. Please check the logs for more information.',
-            'errors' => [],
-        ];
+        switch ($error) {
+            case $error instanceof DomainException:
+                return (int) $error->getCode();
+            default:
+                return self::DEFAULT_STATUS_CODE;
+        }
+    }
 
-        return new JsonResponse($body, self::DEFAULT_STATUS_CODE);
+    private static function buildGenericErrorBody($error): array
```

Comment:
Parameter `$error` is't in use.

Reply:
I know, I left it there for legacy. It wasn't in use in the original implementation either and since I don't know why I didn't want to change it.

I will if everyone agrees to change it, I will.

Reply:
Ohhh, okay!

Reply:
It was done that was for uniformity and also so we didn't have to change the signature in case we wanted to use it in the response.